### PR TITLE
Love Letter

### PR DESCRIPTION
**Pull request by neynt**
There are likely still many corner cases involving targeting yourself, no-target, and better UI.

### CODE REVIEW

File: games/loveletter/rules.js [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+      return false;
+    case 3: {
+      // Compare hands; lower loses.
+      const my_card = game_state.hands[player_id][0];
+      const their_card = game_state.hands[action.target][0];
+      if (my_card > their_card) {
+        kill_player(game_state, action.target);
+        game_state.log.push(
+          `Player ${player_id} uses a 3 to compare hands with Player ${action.target}. ` +
+          `Player ${player_id} is defeated.`);
```
Comment (czinn) [inline games/loveletter/rules.js]: Backwards. If my_card > their_card, action.target is defeated.

File: games/loveletter/rules.js [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+  }
+
+  // Player must be current player
+  if (game_view.player_id !== game_view.current_player) return false;
+  // If the player just played a priest, they're pressing "continue" and their
+  // action will be rewritten to continue the game. All other bets are off.
+  if (game_view.priested_player != null) return true;
+  // Player must have selected card
+  if (!game_view.hands[game_view.player_id].includes(action.card)) return false;
+  if (action.target) {
```
Comment (czinn) [inline games/loveletter/rules.js]: action.target !== undefined. Currently the game always lets you target player 0 because none of the checks run.

**Review by czinn — CHANGES_REQUESTED** [review]

File: games/loveletter/rules.js (hunk 2 above)
Comment (neynt) [reply games/loveletter/rules.js]: Fixed.

File: games/loveletter/rules.js (hunk 1 above)
Comment (neynt) [reply games/loveletter/rules.js]: I think it's just the message that's backwards. Fixed.

**Review by czinn — APPROVED** [review]
There are still issues with self-targeting, but it mostly works.

**Comment by czinn** [comment]
Might want to change "Player i" to the player's username now that that's available.

**Pull request merged**